utils/utils.h

```diff
@@ -262,6 +262,17 @@ static inline bool host_is_lp64(void)
 	return get_elf_class() == ELFCLASS64;
 }
 
+static inline char* has_kernel_opt(char *buf)
+{
+	char *opt = strchr(buf, '@');
+
+	if (opt != NULL)
+		if ((strcmp(opt, "@k") && strcmp(opt, "@kernel")) == 0)
+			return opt;
+
+	return NULL;
+}
```

Comment:
As I said, the filter/trigger string can have multiple actions after module name like `-T abc@kernel,filter`.  So you'd better checking with `strncmp()` and the next character being `,` or `\0`.

Reply:
How about this?
```diff
diff --git a/utils/utils.h b/utils/utils.h
index b56a5a0..dd325d5 100644
--- a/utils/utils.h
+++ b/utils/utils.h
@@ -262,6 +262,22 @@ static inline bool host_is_lp64(void)
        return get_elf_class() == ELFCLASS64;
 }

+static inline char* has_kernel_opt(char *buf)
+{
+       char *opt = strchr(buf, '@');
+
+       if (opt) {
+               int idx = 0;
+               if (!strncmp(opt, "@kernel", 7))
+                       idx = 7;
+               else if (!strncmp(opt, "@k", 2))
+                       idx = 2;
+               if (idx && (opt[idx] == '\0' || opt[idx] == ','))
+                       return opt;
+       }
+       return NULL;
+}
+
 struct uftrace_time_range {
        uint64_t first;
        uint64_t start;
```

Reply:
Looks great. 
Is it okay to apply your code?

Reply:
looks great!
Do you mind if I apply your code in this commit?

Reply:
and also maybe using `strncasecmp` instead of `strncmp` might be more better?

I'm not sure which one to follow, but below here, it compares without case insensivity.

https://github.com/namhyung/uftrace/blob/2aac509417319e17c701a264e5e6e507d6134b36/utils/filter.c#L1000-L1004

Which one should I use?

Reply:
> Do you mind if I apply your code in this commit?

Sure. Please use it.

> and also maybe using strncasecmp instead of strncmp might be more better?

I personally do not want to give multiple usage options, but I'm not sure how @namhyung thinks about it.

Reply:
For consistency, please use `strcasecmp()`.  Also please add a blank line before the if statements.